### PR TITLE
chore: replace black with ruff as the formatter

### PR DESCRIPTION
**Pull request by wood-push-melon**
This pull request tries to:

- replace black with ruff for formatting
- resolve linting and formatting issues

### CODE REVIEW

**Review by BarcoMasile — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -12,33 +12,50 @@ show_missing = true
...
 line-length = 99
-target-version = ["py38"]
+include = ["pyproject.toml", "src/**/*.py", "tests/**/*.py", "lib/charms/openfga_k8s/**/*.py"]
+extend-exclude = ["__pycache__", "*.egg_info"]
+target-version = "py38"
 
-[tool.isort]
-line_length = 99
-profile = "black"
+[too.ruff.lint]
```
Comment (natalian98) [inline pyproject.toml:24]:
```suggestion
[tool.ruff.lint]
```

File: pyproject.toml [hunk 2]
```diff
@@ -12,33 +12,50 @@ show_missing = true
...
+    "E501",
+    "N818"
+]
+per-file-ignores = {"tests/*" = ["D100","D101","D102","D103","D104"]}
 
-# Linting tools configuration
-[tool.flake8]
-max-line-length = 99
-max-doc-length = 99
+[too.ruff.lint.flake8-copyright]
```
Comment (natalian98) [inline pyproject.toml:50]:
```suggestion
[tool.ruff.lint.flake8-copyright]
```

Comment (natalian98) [inline pyproject.toml:50]: Does ruff check for copyright notices? I think we'd need to use `--preview` flag: https://docs.astral.sh/ruff/rules/missing-copyright-notice/

**Review by BarcoMasile — CHANGES_REQUESTED** [review]
Typo in `pyproject.toml` that Natalia caught

**Review by BarcoMasile — APPROVED** [review]

**Review by nsklikas — APPROVED** [review]

**Pull request merged**